siliconcompiler/schema/schema_obj.py

```diff
@@ -78,8 +78,27 @@ def get(self, *keypath, field='value', job=None):
         if not Schema._is_leaf(cfg):
             raise ValueError(f'Invalid keypath {keypath}: get() must be called on a complete keypath')
 
+        if cfg['pernode'] == 'none' and (step is not None or index is not None):
+            raise ValueError(f'step and index are not valid for keypath {keypath}')
```

Comment:
"step and index cannot be used with keypath"?

"step and index are not valid" makes me think that some other combination of step/index might be valid. Ditto for the set() and add() error messages.